common/lib/util/defaults.js

```diff
@@ -102,6 +102,32 @@ Defaults.normaliseOptions = function(options) {
 		options.queueMessages = options.queueEvents;
 	}
 
+	if(options.fallbackHostsUseDefault) {
+		/* fallbackHostsUseDefault and fallbackHosts are mutually exclusive as per TO3k7 */
+		if(options.fallbackHosts) {
+			var msg = 'fallbackHosts and fallbackHostsUseDefault cannot both be set';
+			Logger.logAction(Logger.LOG_ERROR, 'Defaults.normaliseOptions', msg);
+			throw new ErrorInfo(msg, 40000, 400);
+		}
+
+		/* default fallbacks can't be used with custom ports */
+		if(options.port || options.tlsPort) {
+			var msg = 'fallbackHostsUseDefault cannot be set when port or tlsPort are set';
+			Logger.logAction(Logger.LOG_ERROR, 'Defaults.normaliseOptions', msg);
+			throw new ErrorInfo(msg, 40000, 400);
+		}
+
+		/* emit an appropriate deprecation warning */
+		if(options.environment) {
+			Logger.deprecatedWithMsg('fallbackHostsUseDefault', 'There is no longer a need to set this when the environment option is also set since the library will now generate the correct fallback hosts using the environment option.');
+		} else {
+			Logger.deprecated('fallbackHostsUseDefault', 'fallbackHosts: Ably.Defaults.FALLBACK_HOSTS');
+		}
+
+		/* use the default fallback hosts as requested */
+		options.fallbackHosts = Defaults.FALLBACK_HOSTS;
```

Comment:
Maybe this should be moved inside else statement Line 124

Reply:
I don't think so, no.

Reply:
Yeah ofc